src/AppBundle/Controller/DefaultController.php

```diff
@@ -31,4 +31,30 @@ public function contactAction()
         return $this->render('default/contact.html.twig');
 
     }
+    /**
+     * @Route("/catalogue", name="catalogue")
+     */
+    public function catalogueAction()
+    {
+        $em = $this->getDoctrine()->getManager(); //get the database manager
+
+        $plantsSpecification = $em->getRepository('PlantBundle:PlantSpecification')->findAll(); // get the profiles
+
+        return $this->render('default/catalogue.html.twig', array('plants' => $plantsSpecification));
+    }
+    /**
+     * @Route("/catalogue/{$id}", name="showPlant")
+     */
+    public function showPlantSpecificationAction($id)
+    {
+        $plantSpecification =  $this->getDoctrine()
+            ->getRepository('PlantSpecification::class')
+            ->find($id);
+
+        if (!$plantSpecification) {
+            throw $this->createNotFoundException('No plant found for id '.$id);
+        }
+        return $this->render('default/showPlant.html.twig',array('plant' => $plantSpecification));
```

Comment:
for your conveniency, use short array notation.
`[]` == `array()`
so
`array('plant' => $plantSpecification)` == `['plant' => $plantSpecification]`